### DISCUSSION TITLE
Can unique URLs be generated for specific pages such that a user can only access those pages but not the rest of the published vault?

### DISCUSSION THREAD

**Discussion by tsouchlarakis** [ObsidianPublisher/obsidian-github-publisher#182] [Q&A — Plugin] [answered]
Or is it the case that once a user can access the published vault online, they can access every page of that vault?

**Comment by Mara-Li** [comment, marked as answer]
I think it depends of your hosting methods.

As it was send on a github, you can set it private and use a way to build a blog only for x pages. 

With material and mkdocs, it can be possible to remove every navigation (search included) and share a link.

To generate a unique name, you can use template with some Javascript and randomisation ID + a frontmatter key for the generated title.

If you want to just share a private page in the blog with mkdocs, you can use the password plugin and hide the file from the navigation (using Awesome page and hiding in the search)
